### ISSUE TITLE
OFFSET batching strategy should be replaced by a seek strategy

### ISSUE THREAD

**Issue by heldchen** [boxalino/data-integration-magento2#5]
mysql cannot properly optimize for `LIMIT ... OFFSET`. the query performance will become exponentially worse the further the export progresses.  after a certain threshold mysql has additionally to resort to a `filesort` based filtering, further impacting the performance. this happens regardless of the defined batch size, but is affecting the performance more for smaller batchsizes (as more queries are run in total):
```
MariaDB [xxx]> SELECT `s_o`.* FROM `sales_order` AS `s_o` WHERE (s_o.store_id IN ('1', '2') OR s_o.store_id = 0) ORDER BY `s_o`.`entity_id` DESC LIMIT 50 OFFSET 50;
...
50 rows in set (0.003 sec)
```
```
MariaDB [xxx]> SELECT `s_o`.* FROM `sales_order` AS `s_o` WHERE (s_o.store_id IN ('1', '2') OR s_o.store_id = 0) ORDER BY `s_o`.`entity_id` DESC LIMIT 50 OFFSET 500000;
...
50 rows in set (4.389 sec)
```
```
MariaDB [xxx]> SELECT `s_o`.* FROM `sales_order` AS `s_o` WHERE (s_o.store_id IN ('1', '2') OR s_o.store_id = 0) ORDER BY `s_o`.`entity_id` DESC LIMIT 50 OFFSET 2500000;
...
50 rows in set (10.950 sec)
```

### proposal: use seek strategy

seeking by the last processed entity id will provide near linear runtimes no matter the batchsize and current progress. https://www.eversql.com/faster-pagination-in-mysql-why-order-by-with-limit-and-offset-is-slow/ has a good explanation.

for example for the orders this would mean that the query would become:
```
SELECT `s_o`.* FROM `sales_order` AS `s_o` WHERE s_o.entity_id > $LASTORDERID AND (s_o.store_id IN ('1', '2') OR s_o.store_id = 0) ORDER BY `s_o`.`entity_id` ASC  LIMIT 50
```
this allows mysql to make use of the existing index directly:
```
MariaDB [xxx]> SELECT `s_o`.* FROM `sales_order` AS `s_o` WHERE s_o.entity_id > 160550571 AND (s_o.store_id IN ('1', '2') OR s_o.store_id = 0) ORDER BY `s_o`.`entity_id` ASC  LIMIT 50;
...
50 rows in set (0.003 sec)
```
(this is the same position in the export as the example above that took 10.95s)

**Comment by dannegwork** [comment]
Thanks for the suggestion. This can be a future fix for the plugin :) Will look into getting it done by the end of the week

**Comment by dannegwork** [comment]
https://github.com/boxalino/data-integration-magento2/commit/cad7129c8834a4d51cab40596ff37b2258f61bda

the seek strategy was implemented.
it is available as of tag 1.2.4
